controller/engine_controller.go

```diff
@@ -674,15 +674,37 @@ func (m *EngineMonitor) refresh(engine *longhorn.Engine) error {
 
 	needRestore := false
 	if engine.Status.LastRestoredBackup != "" {
-		info, err := client.Info()
+		rsList, err := client.BackupRestoreStatus()
 		if err != nil {
+			logrus.Errorf("failed to get Backup Restore Status: %v", err)
 			return err
 		}
-		// for those engine just restored from backup, info.LastRestored is empty
-		if info.LastRestored != "" {
-			engine.Status.LastRestoredBackup = info.LastRestored
+
+		isRestoring := false
+		lastRestored := ""
+		for _, status := range rsList {
+			if status.IsRestoring {
+				logrus.Errorf("Engine is currently restoring in replica: %v", status.Replica)
+				isRestoring = true
+			}
+		}
+
+		//Engine is not restoring, pick the lastRestored from any replica
+		if !isRestoring {
+			lastRestored = rsList[0].LastRestored
+			for i := 1; i < len(rsList); i++ {
+				if rsList[i].LastRestored != lastRestored {
+					logrus.Warnf("BUG: different lastRestored values in replicas")
```

Comment:
No need for this log if we've returned the error.

Reply:
Will remove it.